lib/active_admin/pundit_adapter.rb

```diff
@@ -57,8 +55,47 @@ def format_action(action, subject)
 
     private
 
+    def policy_target(subject)
+      case subject
+      when nil then resource
+      when Class then subject.new
+      else subject
+      end
+    end
+
+    # This method needed to fallback to our previous policy searching logic.
+    # I.e.: when class name contains `default_policy_namespace` (eg: ShopAdmin)
+    # we should try to search it without namespace (which was a bug)
+    # This fallback might be removed in future versions of ActiveAdmin, so
+    # pundit_adapter search will work consistently with provided namespaces
+    #
+    # See: https://github.com/activeadmin/activeadmin/pull/7144
+    def compat_policy(subject)
+      return unless default_policy_namespace
+
+      target = policy_target(subject)
+
+      if target.class.to_s.include?(default_policy_namespace.to_s.camelize)
+        policy = Pundit.policy(user, target)
+
+        if policy
+          policy_name = policy.class.to_s
+          ns_name = default_policy_namespace.to_s.camelize
+
+          warn "WARN: You have `pundit_policy_namespace` configured, " \
```

Comment:
We have a module for this

```suggestion
          Deprecation.warn "You have `pundit_policy_namespace` configured, " \
```

Also probably worth mentioning the actual value configured?

```suggestion
          Deprecation.warn "You have `pundit_policy_namespace` configured as #{default_policy_namespace}, " \
```